### DISCUSSION TITLE
Return failure if probability is too low

### DISCUSSION THREAD

**Discussion by miesepeter** [SYSTRAN/faster-whisper#416] [Q&A]
I am getting real time spoken user input into my application (short clips under 30 seconds in length). Now sometimes this input cannot be understood (e.g., because of background noise) and faster-whisper just makes up some gibberish. I want to indicate to the user that the input cannot be understood, so that he/she can repeat the input or change the conditions.

Based on the transcribe() method I understand that log_prob_treshold is the setting I am looking for. But I do not understand how a log_prob lower than my threshold could be a condition I can use in my code (as this "failure" info is never returned to me). Am I missing something?

Currently I am using faster-whisper in the following way:
```
    def transcribe(self, audio_bytes):
        if audio_bytes is None:
            return "No user input."
        with torch.no_grad():
            segments, info = self.model.transcribe(
                audio_bytes, beam_size=10, language="en", temperature=0.0, log_prob_threshold=10.0
            )
            print(info)
            text_output = ""
            for segment in segments:
                text_output += segment.text

        return text_output.strip()
```

**Comment by guillaumekln** [comment]
The option `log_prob_threshold` applies for the temperature fallback, but you disabled the fallback with `temperature=0.0`.

If you want to implement some logics based on the log probability, you can read the segment log probability in `segment.avg_logprob`.
